sw/device/lib/crypto/include/hash.h

```diff
@@ -107,19 +137,19 @@ crypto_status_t otcrypto_hash(crypto_const_uint8_buf_t input_message,
  * be returned.
  *
  * @param input_message Input message for extendable output function.
- * @param hash_mode Required extendable output function.
+ * @param cshake_mode Required CSHAKE mode.
  * @param function_name_string NIST Function name string.
  * @param customization_string Customization string for cSHAKE.
  * @param required_output_len Required output length, in bytes.
  * @param[out] digest Output from the extendable output function.
- * @return Result of the xof operation.
+ * @return Result of the CSHAKE xof operation.
  */
-crypto_status_t otcrypto_xof(crypto_const_uint8_buf_t input_message,
-                             xof_mode_t xof_mode,
-                             crypto_const_uint8_buf_t function_name_string,
-                             crypto_const_uint8_buf_t customization_string,
-                             size_t required_output_len,
-                             crypto_uint8_buf_t *digest);
+crypto_status_t otcrypto_xof_cshake(crypto_const_uint8_buf_t input_message,
+                                    xof_mode_t xof_mode,
+                                    crypto_const_uint8_buf_t function_name_string,
+                                    crypto_const_uint8_buf_t customization_string,
+                                    size_t required_output_len,
+                                    crypto_uint8_buf_t *digest);
```

Comment:
The `otcrypto_xof` function is already implemented and has tests that use it (`sw/device/tests/crypto/kmac_verify_functest.c`). So unfortunately, just separating these two functions in thewithout fixing the test and the implementation will break CI. I've got a PR in the works that will separate these, but it's a sizeable change on its own. For now, I'd like to leave `otcrypto_xof` and the `TODO` in `cryptolib_api.md` so that we can get the page up in the meantime.

Reply:
Got it. Thanks. 
I agree to leave the XOF part as is and revisit later.